test/WebJobs.Script.Tests/Workers/Rpc/GrpcWorkerChannelTests.cs

```diff
@@ -688,6 +689,24 @@ public void SendFunctionEnvironmentReloadRequest_WithDirectory()
             Assert.True(envReloadRequest.FunctionAppDirectory == _scriptRootPath);
         }
 
+        [Fact]
+        public async Task SendInvocationRequest_PublishesOutboundEvent_ReceiveInvocationResponse()
+        {
+            await CreateDefaultWorkerChannel();
+            _metricsLogger.ClearCollections();
+            var invocationid = Guid.NewGuid();
+            ScriptInvocationContext scriptInvocationContext = GetTestScriptInvocationContext(invocationid, new TaskCompletionSource<ScriptInvocationResult>());
+            await _workerChannel.SendInvocationRequest(scriptInvocationContext);
+            _testFunctionRpcService.PublishInvocationResponseEvent(invocationid.ToString());
+            await Task.Delay(500);
+            var testWorkerId = _workerId.ToLowerInvariant();
+            var traces = _logger.GetLogMessages();
+            Assert.True(traces.Any(m => string.Equals(m.FormattedMessage, $"InvocationResponse received for invocation id: '{invocationid}'")));
+            Assert.False(string.IsNullOrEmpty(_metricsLogger.LoggedEvents.FirstOrDefault(e => e.Contains($"{string.Format(MetricEventNames.WorkerInvoked, testWorkerId)}_{scriptInvocationContext.FunctionMetadata.Name}"))));
```

Comment:
Same comment as above. We should assert the count of all logged metrics, then verify the exact metrics

Reply:
Updated code here